*: refactor transaction documents (#2088)
pingcap/docs: glossary.md

# Glossary

## A

### ACID

ACID refers to the four key properties of a transaction: atomicity, consistency, isolation, and durability. Each of these properties is described below.

> 

Reviewer: The descriptions of the four properties are very different from the original Chinese. Why is that? Have you reminded the author of this difference? @anotherrachel

Author: Actually, I enrich the descriptions for the Chinese version. The original can be seen here: https://pingcap.com/blog-cn/best-practice-optimistic-transaction/
In English, however, I do refer to some references and make adjustments. It is not translated according to the Chinese as intended. But it is open to discussion. The part focuses on how ACID is achieved in TiDB.

Reviewer: If there's nothing wrong with the original texts, do you think it's a better idea that we try to keep the English documentation consistent with the Chinese documentation?

My concern is that the content might be updated later, if we maintain two different versions of the same content, it might bring extra burden to the author.

But still, it's my own opinion. For this particular content, I don't think it will be updated in the foreseeable future, so it's okay to keep your enriched descriptions. (We'd better merge this PR ASAP.)

- **Atomicity** means that either all the changes of an operation are performed, or none of them are. TiDB ensures the atomicity of the [region](#region) that stores the Primary Key to achieve the atomicity of transactions.

- **Consistency** means that transactions always bring the database from one consistent state to another. In TiDB, data consistency is ensured before writing data to the memory.

> - **Isolation** means that a transaction in process is invisible to other transactions until it completes. This allows concurrent transactions to read and write data while further ensures consistency. TiDB currently supports the isolation level of `REPEATABLE READ`.

Reviewer: ```suggestion
- **Isolation** means that a transaction in process is invisible to other transactions until it completes. This allows concurrent transactions to read and write data without sacrificing consistency. TiDB currently supports the isolation level of `REPEATABLE READ`.
```
